CODING_STANDARDS.md

```diff
@@ -0,0 +1,158 @@
+
+### Line lengths
+
+Lines should be no more than 80 characters.
+
+### Comments
+
+Single line / inline code comments can use the double slash (`//`) style of coding, whereas multi-line code comments should use `/*` as a start and `*/` at the end, with the text indented by 2 spaces, for example:
+
+```cpp
+/*
+  This is a multi-line code comment.
+  It has an indentation of two spaces.
+*/
+```
+
+### Variables classes, and functions
+
+Variables and functions should be descriptive and in "snake case", for example:
+
+```cpp
+void my_function(uint16 variable_name)
+{
+```
+
+Class names should also be "snake case" but should start with an upper-case character. Such as this:
+
+```cpp
+class Buffered_logs
+{
+```
+
+### Constant integers
+
+Constant integers that are used to define elements such as buffer sizes should be defined rather than used directly. This is because the integer could change and uses of it could be missed. For example:
+
+```cpp
+char *buffer= my_malloc(PSI_INSTRUMENT_ME, 1024, MYF(MY_WME));
+
+snprint(buffer, 1024, "%d: %s", integer, text);
+```
+
+Could become:
+
+```cpp
+constexpr int buffer_size= 1024;
+char *buffer= my_malloc(PSI_INSTRUMENT_ME, buffer_size, MYF(MY_WME));
+
+snprint(buffer, buffer_size, "%d: %s", integer, text);
+```
+
+Alternatively the integer can be defined using an `enum` or `#define`.
+
+### Spacing
+
+#### Whitespace
+
+* Lines should not have any trailing whitespace.
+* There should not be any trailing blank lines at the end of a file.
+* Line endings are POSIX style (`\n`).
+* Two spaces for each indentation level, not tabs.
+
+#### Pointers
+
+The `*` of a pointer should be on the side of the variable name such as:
+
+```cpp
+void my_function(THD *thd)
+{
+```
+
+As yet there is no standard as to whether the `*` in a casting should have a space or not. Both of these are valid:
+
+```cpp
+name= (const char*) db_name;
+name= (const char *) db_name;
+```
+
+#### Equals
+
+When setting a variable there should be no space on the left hand side and one space on the right hand side. For example:
+
+```cpp
+number= 123;
+```
+
+#### Function variables
+
+There should be a space after each comma in a definition and usage of a function. For example:
+
+```cpp
+my_function(thd, db_name);
+```
+
+### Types
+
+In general the usage of types such as `char`, `int` and `long` should be discouraged but there are shortend versions of the unsigned variants available for these in `my_global.h`. They can be different sizes across platforms and `char` can be either unsigned or signed depending on platform, and therefore are not portable. Instead these should be used as appropriate:
+
+* 8-bit signed / unsigned int -> `int8` / `uint8`
+* 16-bit signed / unsigned int -> `int16` / `uint16`
+* 32-bit signed / unsigned int -> `int32` / `uint32`
+* 64-bit signed / unsigned int -> `int64` / `uint64`
```

Comment:
I think that `size_t` and `ptrdiff_t` deserve to be mentioned explicitly. Their size is the integer register width, that is, 32 or 64 bits.